pkg/common/utils/gang_utils.go

```diff
@@ -123,24 +143,47 @@ func GetTaskGroupsFromAnnotation(pod *v1.Pod) ([]v1alpha1.TaskGroup, error) {
 	return taskGroups, nil
 }
 
-func GetPlaceholderTimeoutParam(pod *v1.Pod) (int64, error) {
+func GetSchedulingPolicyParam(pod *v1.Pod) *SchedulingPolicyParameters {
+	timeout := int64(0)
+	style := constants.SchedulingPolicyStyleParamDefault
+	schedulingPolicyParams := NewSchedulingPolicyParameters(timeout, style)
 	param, ok := pod.Annotations[constants.AnnotationSchedulingPolicyParam]
 	if !ok {
-		return 0, nil
+		return schedulingPolicyParams
 	}
 	params := strings.Split(param, constants.SchedulingPolicyParamDelimiter)
+	var timeoutErr bool
+	var styleErr bool
+	var err error
+	var styleExists bool
 	for _, p := range params {
-		timeoutParam := strings.Split(p, "=")
-		if timeoutParam[0] == constants.SchedulingPolicyTimeoutParam {
-			if len(timeoutParam) != 2 {
-				return 0, fmt.Errorf("unable to parse timeout value from annotation")
+		param := strings.Split(p, "=")
+		if param[0] == constants.SchedulingPolicyTimeoutParam {
+			if len(param) != 2 {
+				timeoutErr = true
+				log.Logger().Info("Unable to parse timeout value from annotation", zap.String("namespace", pod.Namespace), zap.String("name", pod.Name), zap.Int64("Using Placeholder timeout: ", timeout), zap.String("Placeholder timeout passed in annotation: ", p))
+			}
```

Comment:
this is dup code, we could move the check before the parameter key check.
If len(param) != 2, we can simply skip it with a logging message:  "skipping malformed scheduling policy parameter: xxxx". We can still ensure the default values are used. With this, we can remove `timeoutErr`, `styleErr`, `styleExists` bool values and further simplify the code.